tests/e2e/upgradecluster/upgradecluster_test.go

```diff
@@ -0,0 +1,460 @@
+				for _, pod := range pods {
+					if strings.Contains(pod.Name, "helm-install") {
+						g.Expect(pod.Status).Should(Equal("Completed"), pod.Name)
+					} else {
+						g.Expect(pod.Status).Should(Equal("Running"), pod.Name)
+					}
+				}
+			}, "420s", "5s").Should(Succeed())
+			_, _ = e2e.ParsePods(kubeConfigFile, true)
+		})
+
+		It("Verifies ClusterIP Service", func() {
+			_, err := e2e.DeployWorkload("clusterip.yaml", kubeConfigFile, false)
+			if err != nil {
+				fmt.Println("Cluster IP manifest not deployed", err)
+				Expect(err).NotTo(HaveOccurred())
+			}
+			Eventually(func(g Gomega) {
+				cmd := "kubectl get pods -o=name -l k8s-app=nginx-app-clusterip --field-selector=status.phase=Running --kubeconfig=" + kubeConfigFile
+				res, err := e2e.RunCommand(cmd)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(res).Should(ContainSubstring("test-clusterip"))
+			}, "240s", "5s").Should(Succeed())
+
+			clusterip, _ := e2e.FetchClusterIP(kubeConfigFile, "nginx-clusterip-svc")
+			cmd := "\"curl -L --insecure http://" + clusterip + "/name.html\""
+			fmt.Println(cmd)
+			for _, nodeName := range serverNodenames {
+				Eventually(func(g Gomega) {
+					res, err := e2e.RunCmdOnNode(cmd, nodeName)
+					g.Expect(err).NotTo(HaveOccurred())
+					fmt.Println(res)
+					g.Expect(res).Should(ContainSubstring("test-clusterip"))
+				}, "120s", "10s").Should(Succeed())
+			}
+		})
+
+		It("Verifies NodePort Service", func() {
+			_, err := e2e.DeployWorkload("nodeport.yaml", kubeConfigFile, false)
+			if err != nil {
+				fmt.Println("NodePort manifest not deployed", err)
+				Expect(err).NotTo(HaveOccurred())
+			}
+			for _, nodeName := range serverNodenames {
+				node_external_ip, _ := e2e.FetchNodeExternalIP(nodeName)
+				cmd := "kubectl get service nginx-nodeport-svc --kubeconfig=" + kubeConfigFile + " --output jsonpath=\"{.spec.ports[0].nodePort}\""
+				nodeport, err := e2e.RunCommand(cmd)
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(func(g Gomega) {
+					cmd := "kubectl get pods -o=name -l k8s-app=nginx-app-nodeport --field-selector=status.phase=Running --kubeconfig=" + kubeConfigFile
+					res, err := e2e.RunCommand(cmd)
+					g.Expect(err).NotTo(HaveOccurred())
+					g.Expect(res).Should(ContainSubstring("test-nodeport"), "nodeport pod was not created")
+				}, "240s", "5s").Should(Succeed())
+
+				cmd = "curl -L --insecure http://" + node_external_ip + ":" + nodeport + "/name.html"
+				fmt.Println(cmd)
+				Eventually(func(g Gomega) {
+					res, err := e2e.RunCommand(cmd)
+					g.Expect(err).NotTo(HaveOccurred())
+					fmt.Println(res)
+					g.Expect(res).Should(ContainSubstring("test-nodeport"))
+				}, "240s", "5s").Should(Succeed())
+			}
+		})
+
+		It("Verifies LoadBalancer Service", func() {
+			_, err := e2e.DeployWorkload("loadbalancer.yaml", kubeConfigFile, false)
+			if err != nil {
+				fmt.Println(err, "loadbalancer manifest not deployed")
+				Expect(err).NotTo(HaveOccurred())
+			}
```

Comment:
Can be combined into one line
```suggestion
			Expect(err).NotTo(HaveOccurred(), "loadbalancer manifest not deployed")
```